File: d5/solve.py
passes = open('input.txt', 'r').read().strip().split('\n')

def parse(passport: str) -> int:
  row = 0
  for c in passport[:7]:
    row = 2*row + (1 if c == 'B' else 0)
  col = 0
  for c in passport[7:]:
    col = 2*col + (1 if c == 'R' else 0)

  return row * 8 + col

passNum = [parse(passport) for passport in passes]
print("maxID", max(passNum))

File: d5/solve.cpp
#include <bits/stdc++.h>
using namespace std;

struct pass {
  int row;
  int col;
};

char buf[20];

int main() {
  int i=0;
  pass passes[2000];

  freopen("input.txt", "r", stdin);

  int mx = 0;
  while (scanf("%s\n", buf) != EOF) {
  // for (int i=0; i<10; i++) {
    // printf("%s\n", buf);
    int row = 0;
    for (int i=0; i<7; i++) {
      row = 2*row + (buf[i] == 'B');
    }
    int col = 0;
    for (int i=7; i<10; i++) {
      col = 2*col + (buf[i] == 'R');
    }
    passes[i] = {row, col};
    mx = max(mx, row*8 + col);
  }
  printf("%d\n", mx);
  return 0;
}